### PR TITLE
Release the python REST client for REST VERSION 1.9

### PR DESCRIPTION
**Pull request by tlewis-ps**
This accompanies the PurityFB//3.0.0 release.

- New Features

-- Zero Touch Provisioning
A new process for configuring a FlashBlade system after completing the hardware
installation and initial upgrade/bootstrap no longer requires a direct connection to
the serial console to provision the FlashBlade. ZTP simplifies provisioning of the
system by allowing the process to be completed remotely by automatically obtaining
an IP address (per entry point) over its management port via DHCP. After an initial
setup command on the console, you can use the REST token ("PURESETUP") to access the
array with a set of allowable APIs to perform basic configuration of the array and
static management network. The REST endpoints "verification" and "finalization"
respectively verifies and completes the provisioning process. Once complete, the
"PURESETUP" token becomes invalidated and DHCP is terminated.

- New Endpoints

-- Arrays

The following endpoints have been added to view the performance metrics of the NFS
protocol and all replication:
GET arrays/nfs-specific-performance
GET arrays/performance/replication

-- Array Connections

The following endpoints have been added to create a connection between two FlashBlade
arrays. In addition, you can view the replication performance data and modify the
connection configuration:
GET | POST | PATCH | DELETE array-connections
GET | POST array-connections/connection-key
GET array-connections/path
GET array-connections/performance/replication

-- Audits

The following endpoint has been added to display the full audit activity of the
FlashBlade array:
GET audits

-- Bucket Replica Links

The following endpoint has been added to manage replica links for object replication:
GET | POST | PATCH | DELETE bucket-replica-links

-- File Systems

The following endpoints have been added to manage a replica link configuration of a file
system:
GET | POST | PATCH file-system-replica-links
GET | POST | DELETE file-system-replica-links/policies
GET file-system-replica-links/transfer

-- File System Snapshots

The following endpoint has been added to display the details of a file system snapshot
replicating to a target array:
GET | DELETE file-system-snapshots/transfer

-- Keytabs

The following endpoints have been added to manage Kerberos keytab files:
GET | DELETE keytabs
GET keytabs/download
POST keytabs/upload

-- Object Store

The following endpoint has been added to manage the remote credentials used for object
replication:
GET | POST | PATCH | DELETE object-store-remote-credentials

-- Policies

The following endpoint has been added to manage file system snapshot policies for
replication:
GET | POST | DELETE policies/file-system-replica-links

-- SNMP

The following endpoints have been added to manage SNMP agents and managers:
GET | PATCH snmp-agents
GET snmp-agents/mib
GET | POST | PATCH | DELETE snmp-managers
GET snmp-managers/test

-- Targets

The following endpoints have been added to manage configuration information of S3 targets
for object replications:
GET | POST | PATCH | DELETE targets
GET targets/performance/replication

-- Zero Touch Provisioning

The following endpoints have been added to manage setup of ZTP:
GET setup/validation
PATCH setup/finalization

- Enhanced Endpoints

-- Alert Watchers

The following endpoint has been enhanced to manage the alert watcher’s threshold of alert
notifications. The minimum alert thresholds can be set to info, warning, critical:
GET | POST | PATCH alert-watchers

-- Directory Services

The following endpoint has been updated to allow management logins and user credentials
to be modified:
GET | PATCH  directory-services

New examples have been added for all of the newly supported options.

### CODE REVIEW

File: PKG-INFO [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Metadata-Version: 1.8.0
+Metadata-Version: 1.9
```
Comment (azornberg-ps) [inline PKG-INFO]: can you make this "1.9.0"? it makes it easier if we find a bug in the client and need to update it

File: PKG-INFO [hunk 2]
```diff
@@ -1,5 +1,5 @@
...
 Name: purity_fb
-Version: 1.8.0
-Summary: A python client for Purity//FB REST API.
+Version: 1.9
```
Comment (azornberg-ps) [inline PKG-INFO]: can you make this "1.9.0"? it makes it easier if we find a bug in the client and need to update it

File: README.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
-# purity-fb
-Client for Purity//FB REST API (1.0 - 1.7), developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
+# Pure Storage FlashBlade REST 1.9 Python SDK Documentation
+Pure Storage FlashBlade REST 1.9 Python SDK, developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
```
Comment (azornberg-ps) [inline README.md]: can we still note that it can be used for versions "(1.0 - 1.9) somewhere?

File: setup.py [hunk 4]
```diff
@@ -3,7 +3,7 @@
 """
     Purity//FB REST Client
 
-    Client for Purity//FB REST API (1.0 - 1.8), developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
+    Client for Purity//FB REST API (1.0 - 1.9), developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
```
Comment (azornberg-ps) [inline setup.py]: should this description be updated for the client renaming the way that the others were?

File: setup.py [hunk 5]
```diff
@@ -15,7 +15,7 @@
 from setuptools import setup, find_packages
 
 NAME = "purity-fb"
-VERSION = "1.8.0"
+VERSION = "1.9"
```
Comment (azornberg-ps) [inline setup.py]: this is the package version so can we make it "1.9.0" in case we need to fix bugs in the client?

File: setup.py [hunk 6]
```diff
@@ -28,14 +28,14 @@
...
+    description="Pure Storage FlashBlade REST 1.9 Python SDK",
     author_email="info@purestorage.com",
     url="",
     keywords=["Swagger", "Purity//FB REST Client"],
     install_requires=REQUIRES,
     packages=find_packages(),
     include_package_data=True,
     long_description="""\
-    Client for Purity//FB REST API (1.0 - 1.8), developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
+    Pure Storage FlashBlade REST 1.9 Python SDK, developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).
```
Comment (azornberg-ps) [inline setup.py]: can we still note somewhere that this can be used for REST API versions (1.0 - 1.9)?

**Review by azornberg-ps — CHANGES_REQUESTED** [review]

File: setup.py (hunk 6 above)
Comment (tlewis-ps) [reply setup.py]: This comment string gets stamped out at the top of every generated python file. Should I be updating all of them?

How about:

> Pure Storage FlashBlade REST 1.9 Python SDK. Compatible with REST API versions 1.0 - 1.9. Developed by [Pure Storage, Inc](http://www.purestorage.com/). Documentations can be found at [purity-fb.readthedocs.io](http://purity-fb.readthedocs.io/).

Comment (azornberg-ps) [reply setup.py]: ehh you probably don't need to update all of them. I just figured the setup and top level stuff was more likely to be read

**Review by azornberg-ps — APPROVED** [review]

**Pull request merged**